### ISSUE TITLE
[Vala] Generated file in wrong subdirectory

### ISSUE THREAD

**Issue by jangernert** [mesonbuild/meson#3147]
I'm generating a `Constants.vala` file like this
```
intermediate = configure_file(
	input: 'Constants.vala.in',
	output: 'Constants.intermediate.vala',
	configuration: conf)

constants = vcs_tag(
	input : intermediate,
	output : 'Constants.vala',
	command : ['git', 'show-ref', '-s', 'refs/heads/master'],
	replace_string: '#GIT_SHA1#')
```

That file is saved in the root build directory.
Later I feed that file as part of my vala files into an `library()` command.

```
vala_sources = [
  constants,
  'src/FavIcon.vala',
  'src/FeedReader.vala',
  (...)
]
```

The `Constants.c` file is then generated in `build/FeedReader@sha/build/`. However it is expected to be located at `build/FeedReader@sha/`
```
cc: error: FeedReader@sha/Constants.c: No such file or directory
```
Copying the file into the parent directory makes the build complete without further issues.

**Comment by nirbheek** [comment]
Which version of meson are you using?

**Comment by jangernert** [comment]
Oh, sorry forgot to mention that. I'm using `0.44.0`.
The full build file is [here](https://github.com/jangernert/FeedReader/blob/meson-3/meson.build)

**Comment by nirbheek** [comment]
This was fixed in https://github.com/mesonbuild/meson/pull/3101 and released as 0.44.1, please try that.

**Comment by jangernert** [comment]
Indeed it does fix it. Since the update is only 8 days old, even fedora hasn't caught up.
Thx
